Library/Homebrew/linkage_checker.rb

```diff
@@ -83,6 +90,13 @@ def check_undeclared_deps
       next true if Formula[name].bin.directory?
       @brewed_dylibs.keys.map { |x| x.split("/").last }.include?(name)
     end
+    missing = Set.new
+    @broken_dylibs.each_value do |value|
+      value.each do |str|
+        missing << dylib_to_dep(str) if str.start_with?("#{HOMEBREW_PREFIX}/opt", HOMEBREW_CELLAR)
```

Comment:
You could check if this matches the `dylib_to_dep` regex above instead of having another `start_with`